src/screens/LegalRights/VideoPage/index.tsx

```diff
@@ -1,12 +1,69 @@
+import { useEffect, useRef, useState } from 'react';
 import { Pressable, ScrollView, Text, View } from 'react-native';
 import { Video } from 'expo-av';
+import supabase from '@/supabase/createClient';
 import { styles } from './styles';
 
-export default function VideoPage() {
+export default function VideoPage(testProp: any) {
+  // var for array of all the pages for the current language
+  const [preaData, setPreaData] = useState([
+    {
+      id: 'string',
+      is_short_answer: true,
+      page_number: 0,
+      parent_id: 'string',
+      short_answer: 'string',
+      spanish: false,
+      survey: 'string',
+      video_id: 'Section Title 1',
+    },
+  ]);
+  const [index, setIndex] = useState(Infinity); // index of current page in full array of pages; have to set to infinite or else if the first page (actually index 0) is pressed, the videopage wont update
+  const [language, setLanguage] = useState('english'); // which language associated to array of pages
+
+  const videoLinkRef = useRef(
+    'https://d23dyxeqlo5psv.cloudfront.net/big_buck_bunny.mp4',
+  );
+
+  //
+  const [renderTrigger, setRenderTrigger] = useState(0);
+
+  const nextPage = () => {
+    setIndex(prevIndex => Math.min(prevIndex + 1, preaData.length - 1)); // next index in array of pages
+  };
+
+  const prevPage = () => {
+    setIndex(prevIndex => Math.max(prevIndex - 1, 0)); // previous index in array of pages
+  };
+
+  // read in all the variables passed in from legal rights page; only once when initally rendered
+  useEffect(() => {
+    setPreaData(testProp['route']['params'][0]);
+    setIndex(testProp['route']['params'][1]);
+    setLanguage(testProp['route']['params'][2]);
+  }, []);
+
+  useEffect(() => {
+    const fetchVideoLink = async () => {
+      let response = supabase.storage
```

Comment:
in general, supabase queries should be abstracted away from the screens. Write this function in `/supabase/generalQueries`, or, it might be a good idea to make a dedicated file for storage queries as well